File: src/builder/java/builder/LolChampionBuilder.java
package builder.java.builder;

import builder.java.champion.Champion;
import static builder.java.champion.Champion.*;
import builder.java.champion.LolChampion;

/**
 * Created by kvirus on 2019/6/13 18:55
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public class LolChampionBuilder implements Builder {
    LolChampion instance = new LolChampion();

    @Override
    public Builder setNickname(String nickname) {
        instance.nickname = nickname;
        return this;
    }

    @Override
    public Builder setSlogon(String slogon) {
        instance.slogon = slogon;
        return this;
    }

    @Override
    public Builder setContinent(String continent) {
        instance.continent = continent;
        return this;
    }

    @Override
    public Builder setType(ChampionType type) {
        instance.type = type;
        return this;
    }

    @Override
    public Champion build() {
        return instance;
    }
}

File: src/builder/kotlin/builder/LolChampionBuilder.kt
package builder.kotlin.builder

import builder.kotlin.champion.Champion
import builder.kotlin.champion.ChampionType
import builder.kotlin.champion.LolChampion

/**
 * Created by kvirus on 2019/6/13 19:47
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
class LolChampionBuilder: Builder {
    var instance: LolChampion = LolChampion()

    override fun setNickname(nickname: String): Builder {
        instance.nickname = nickname
        return this
    }

    override fun setSlogon(slogon: String): Builder {
        instance.slogon = slogon
        return this
    }

    override fun setContinent(continent: String): Builder {
        instance.continent = continent
        return this
    }

    override fun setType(type: ChampionType): Builder {
        instance.type = type
        return this
    }

    override fun build(): Champion {
        return instance
    }
}